x-pack/plugins/ml/public/file_datavisualizer/components/import_summary/import_summary.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React from 'react';
+
+import {
+  EuiSpacer,
+  EuiDescriptionList,
+  EuiCallOut,
+  EuiAccordion,
+} from '@elastic/eui';
+
+export function ImportSummary({
+  index,
+  indexPattern,
+  ingestPipelineId,
+  docCount,
+  importFailures,
+}) {
+  const items = createDisplayItems(
+    index,
+    indexPattern,
+    ingestPipelineId,
+    docCount,
+    importFailures
+  );
+
+  return (
+    <React.Fragment>
+      <EuiCallOut
+        title="Import complete"
+        color="success"
+        iconType="check"
+      >
+        <div className="import-summary-list">
+          <EuiDescriptionList
+            type="column"
+            listItems={items}
+          />
+        </div>
+      </EuiCallOut>
+
+      {(importFailures && importFailures.length > 0) &&
+        <React.Fragment>
+          <EuiSpacer size="m" />
+          <EuiCallOut
+            title="Some documents could not be imported"
```

Comment:
I know it's unlikely, but is it possible that the file import could only contain one line? If so, should do a plural check on `documents` in the text here.

Reply:
i've been unable to produce a situation where 1 out of 1 document document fails to be ingested.
With 1 line in a semi structured text file, the file structure endpoint fails to produce results.
I'm going to leave this as is and i don't want to change it without testing.

Reply:
Yes, the backend needs to see 3 lines with timestamps in the same format to detect semi-structured text. Two complete messages plus the last one that gets ignored by the structure finder in case it’s partial (but proves the one above is complete).